tests/ndc/replication_task_batching_test.go

```diff
@@ -0,0 +1,337 @@
+		{ClusterName: clusterName[0]},
+		{ClusterName: clusterName[1]},
+	}
+	_, err := passiveFrontend.RegisterNamespace(context.Background(), &workflowservice.RegisterNamespaceRequest{
+		Namespace:                        s.namespace.String(),
+		IsGlobalNamespace:                true,
+		Clusters:                         replicationConfig,
+		ActiveClusterName:                clusterName[0],
+		WorkflowExecutionRetentionPeriod: durationpb.New(1 * time.Hour * 24),
+	})
+	s.Require().NoError(err)
+	// Wait for namespace cache to pick the change
+	time.Sleep(2 * tests.NamespaceCacheRefreshInterval)
+
+	descReq := &workflowservice.DescribeNamespaceRequest{
+		Namespace: s.namespace.String(),
+	}
+	resp, err := passiveFrontend.DescribeNamespace(context.Background(), descReq)
+	s.Require().NoError(err)
+	s.Require().NotNil(resp)
+	s.namespaceID = namespace.ID(resp.GetNamespaceInfo().GetId())
+
+	s.logger.Info("Registered namespace", tag.WorkflowNamespace(s.namespace.String()), tag.WorkflowNamespaceID(s.namespaceID.String()))
+}
+
+func (s *NDCReplicationTaskBatchingTestSuite) GetReplicationMessagesMock() (*adminservice.StreamWorkflowReplicationMessagesResponse, error) {
+	task := <-s.standByReplicationTasksChan
+	taskID := atomic.AddInt64(&s.standByTaskID, 1)
+	task.SourceTaskId = taskID
+	tasks := []*replicationspb.ReplicationTask{task}
+
+	replicationMessage := &repicationpb.WorkflowReplicationMessages{
+		ReplicationTasks:       tasks,
+		ExclusiveHighWatermark: taskID + 1,
+	}
+
+	return &adminservice.StreamWorkflowReplicationMessagesResponse{
+		Attributes: &adminservice.StreamWorkflowReplicationMessagesResponse_Messages{
+			Messages: replicationMessage,
+		},
+	}, nil
+}
+
+func (s *NDCReplicationTaskBatchingTestSuite) createHistoryEventReplicationTaskFromHistoryEventBatch(
+	namespaceId string,
+	workflowId string,
+	runId string,
+	events []*historypb.HistoryEvent,
+	newRunEvents []*historypb.HistoryEvent,
+	versionHistoryItems []*historyspb.VersionHistoryItem,
+) *replicationspb.ReplicationTask {
+	eventBlob, err := s.serializer.SerializeEvents(events, enumspb.ENCODING_TYPE_PROTO3)
+	var newRunEventBlob *commonpb.DataBlob
+	if newRunEvents != nil {
+		newRunEventBlob, err = s.serializer.SerializeEvents(newRunEvents, enumspb.ENCODING_TYPE_PROTO3)
+		s.NoError(err)
+	}
+	s.NoError(err)
+	taskType := enumsspb.REPLICATION_TASK_TYPE_HISTORY_V2_TASK
+	replicationTask := &replicationspb.ReplicationTask{
+		TaskType: taskType,
+		Attributes: &replicationspb.ReplicationTask_HistoryTaskAttributes{
+			HistoryTaskAttributes: &replicationspb.HistoryTaskAttributes{
+				NamespaceId:         namespaceId,
+				WorkflowId:          workflowId,
+				RunId:               runId,
+				VersionHistoryItems: versionHistoryItems,
+				Events:              eventBlob,
+				NewRunEvents:        newRunEventBlob,
+			}},
+	}
+	return replicationTask
+}
+
+func (s *NDCReplicationTaskBatchingTestSuite) eventBatchesToVersionHistory(
```

Comment:
there are 2 eventBatchesToVersionHistory in the PR. duplicated?

Reply:
consolidated.